polygon-validator-signer-balance-bot/src/agent.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { FindingType, FindingSeverity, Finding, ethers, HandleBlock, createBlockEvent } from "forta-agent";
+import { provideHandleTransaction } from "./agent";
+import { POLYGON_VALIDATOR_SIGNER_ADDRESS } from "./constants";
+
+// let MINIMUM_THRESHOLD_BIG = ethers.BigNumber.from(MINIMUM_THRESHOLD)
+describe("POLYGON-VALIDATOR-SIGNER BOT TEST SUITE", () => {
+  let handleBlock: HandleBlock;
+
+  const mockEthersProvider = {
+    getBalance: jest.fn(),
+  } as any;
+
+  const blockEvent = createBlockEvent({
+    block: { hash: "0xa", number: 1 } as any,
+  });
+
+  beforeAll(() => {
+    handleBlock = provideHandleTransaction(mockEthersProvider);
+  });
+
+  it("returns  findings if balance is above threshold", async () => {
+    const balance = ethers.BigNumber.from("2000000000000000000");
+    const formatThresholdToBigNumber = ethers.BigNumber.from("1000000000000000000");
+    mockEthersProvider.getBalance.mockReset();
+    mockEthersProvider.getBalance.mockReturnValueOnce(balance);
+    const findings = await handleBlock(blockEvent);
+
+    expect(mockEthersProvider.getBalance).toHaveBeenCalledTimes(1);
+    expect(mockEthersProvider.getBalance).toHaveBeenCalledWith(
+      POLYGON_VALIDATOR_SIGNER_ADDRESS,
+      blockEvent.blockNumber - 1
+    );
+    expect(findings).toStrictEqual([
+      Finding.fromObject({
+        name: "Account balance greater than threshold!",
+        description: `Account balance (${balance}) above threshold (${formatThresholdToBigNumber})`,
+        alertId: "POLYGON-VALIDATOR-SIGNER-1",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        protocol: "POLYGON-VALIDATOR-SIGNER-BOT",
+        metadata: {
+          balance: balance.toString(),
+        },
+      }),
+    ]);
+  });
+
+  it("returns empty array while account balance is greater than threshold", async () => {
```

Comment:
This shouldn't rely on the state of a different unit test. If you want to run a test in which you're having the bot check two distinct blocks, you can do so in the same unit test by calling the `handleBlock()` more than once in the same unit test with different blocks, and check the return values after each.

Along the same lines, the state should be cleared in between tests, and therefore, this test should fail because it is relying on the bot having memory of the actions taken place in the previous test.